### PR TITLE
Fix next functionality in  userController

### PR DESCRIPTION
**Pull request by eugleenyc**

### CODE REVIEW

File: Server/controllers/CrudController.ts [hunk 1]
```diff
@@ -1,8 +1,8 @@
-import { Request, Response } from 'express';
+import express, { Request, Response } from 'express';
 
 export abstract class CrudController {
     public abstract create(req: Request, res: Response): void;
-    public abstract read(req: Request, res: Response): void;
+    public abstract read(req: Request, res: Response, next:express.NextFunction): void;
```
Comment (eugleenyc) [inline Server/controllers/CrudController.ts:5]: Need to add this to reflect in User.ts

File: Server/controllers/User/User.ts [hunk 2]
```diff
@@ -18,13 +18,15 @@ export class UserController extends CrudController {
         throw new Error("Method not implemented.");
     }
 
-    public async read(req: Request<import("express-serve-static-core").ParamsDictionary>, res: Response): Promise<AlgoList>   {
+    public async read(req: Request<import("express-serve-static-core").ParamsDictionary>, res: Response, next:express.NextFunction):Promise<void>   {
```
Comment (eugleenyc) [inline Server/controllers/User/User.ts:21]: Changed this promise to Void. This should not return anything as it should next back to the userRouter

File: Server/controllers/User/User.ts [hunk 3]
```diff
@@ -18,13 +18,15 @@ export class UserController extends CrudController {
...
             console.log('in user controller')
             const qAlgos  = 'SELECT * FROM "public"."public.algo-questions" LIMIT 100';
             const homeQuery = await db.query(qAlgos);
             const details = homeQuery.rows;
-             console.log(details);
-            return res.locals = details ;
+            console.log(details);
+            res.locals.details = details;
+            next();
```
Comment (eugleenyc) [inline Server/controllers/User/User.ts:28]: Next works now!

Comment (crumblepie) [reply Server/controllers/User/User.ts:28]: Great debugging, Eugene!

File: Server/controllers/User/User.ts (hunk 2 above)
Comment (crumblepie) [reply Server/controllers/User/User.ts:21]: Agreed.

**Review by crumblepie — APPROVED** [review]

**Pull request merged**